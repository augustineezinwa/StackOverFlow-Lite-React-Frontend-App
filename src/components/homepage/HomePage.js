import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import QuestionCard from './QuestionCard';
import { fetchQuestions } from '../../actions/fetchQuestionsActions';

export class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  componentDidMount() {
    const { fetchAllQuestions } = this.props;
    fetchAllQuestions();
  }

  render() {
    const { questions } = this.props;

    return (
      <Fragment>
        <div id="pageDisplay">
          <div className="container image-background" style={{ margin: '0 auto' }}>

            <div className="row">

              <div className="col mt-10">
                <h1 style={{ color: '#FFFFFF', textAlign: 'center' }}>Welcome to StackOverFlow-Lite</h1>
              </div>

            </div>

          </div>

          <div className="container dashboardfooter " id="dashBoardTitle"><h3>Trending Questions</h3></div>

          <div id="questionsDisplay">

            {
              !questions.length && (
                <div className="container">
                  <div className="row no-questions">

                    <div className="alignCardWidth">
                      <div className="card">
                        <div className="container">
                          <div className="row mt-4 pd-1">
                            <div className="col-2">
                              <div className="symbol-display">
                                <div className="alignSymbol">!</div>
                              </div>

                            </div>
                            <div className="col-5">
                              <div className="question">No Questions yet!  &nbsp; Refresh page  </div>

                            </div>
                          </div>

                          <div className="col" style={{ textAlign: 'right' }}>
                            <span />
                            <span />
                            <a href="/">
                              <button type="answer">Refresh</button>
                            </a>

                          </div>

                        </div>
                      </div>
                    </div>

                  </div>
                </div>
              )
            }
            {
              !!questions.length && (
                <div
                  id="main-fish"
                  className="maincont"
                   >
                  {questions.map(x => (
                    <div className="row" key={x.id}>
                      <QuestionCard
                        key={x.id}
                        questionTitle={x.questionTitle}
                        questionId={x.id}
                        answerNumber={x.numberOfAnswers}
                        totalUpVotes={x.upvotes}
                        totalDownVotes={x.downvotes} />
                    </div>
                  ))}
                </div>
              )
            }


          </div>
        </div>
      </Fragment>
    );
  }
}


export const mapStateToProps = state => ({
  questions: state.questions.data
});

const mapActionsToProps = {
  fetchAllQuestions: fetchQuestions
};


HomePage.propTypes = {
  fetchAllQuestions: PropTypes.func.isRequired,
  questions: PropTypes.object.isRequired,
};


export default connect(mapStateToProps, mapActionsToProps)(HomePage);
