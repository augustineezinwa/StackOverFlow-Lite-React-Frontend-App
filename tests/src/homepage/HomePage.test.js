import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../../../src/components/homepage/HomePage';

test('App tests', () => {
  const homePageComponent = shallow(<HomePage />);
  expect(homePageComponent).toMatchSnapshot();
});
