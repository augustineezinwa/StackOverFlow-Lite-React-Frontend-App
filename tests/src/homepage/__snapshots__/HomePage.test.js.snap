// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests 1`] = `
<Fragment>
  <div
    id="pageDisplay"
  >
    <div
      className="container image-background"
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col mt-10"
        >
          <h1
            style={
              Object {
                "color": "#FFFFFF",
                "textAlign": "center",
              }
            }
          >
            Welcome to StackOverFlow-Lite
          </h1>
        </div>
      </div>
    </div>
    <div
      className="container dashboardfooter "
      id="dashBoardTitle"
    >
      <h3>
        Trending Questions
      </h3>
    </div>
    <div
      id="questionsDisplay"
    >
      <div
        className="container"
      >
        <div
          className="row no-questions"
        >
          <div
            className="alignCardWidth"
          >
            <div
              className="card"
            >
              <div
                className="container"
              >
                <div
                  className="row mt-4 pd-1"
                >
                  <div
                    className="col-2"
                  >
                    <div
                      className="symbol-display"
                    >
                      <div
                        className="alignSymbol"
                      >
                        !
                      </div>
                    </div>
                  </div>
                  <div
                    className="col-5"
                  >
                    <div
                      className="question"
                    >
                      No Questions yet!  Â  Refresh page  
                    </div>
                  </div>
                </div>
                <div
                  className="col"
                  style={
                    Object {
                      "textAlign": "right",
                    }
                  }
                >
                  <span />
                  <span />
                  <a
                    href="/"
                  >
                    <button
                      type="answer"
                    >
                      Refresh
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
